stdlib/src/builtin/string.mojo

```diff
@@ -2109,3 +2198,76 @@ fn _calc_format_buffer_size[type: DType]() -> Int:
         return 64 + 1
     else:
         return 128 + 1  # Add 1 for the terminator
+
+
+# ===----------------------------------------------------------------------===#
+# Format method structures
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct _FormatCurlyEntry:
+    """
+    Internally used by the `format()` method.
+
+    Purpose of the variant types:
+
+
+    - `Int` for manual indexing of *args
+
+        ```mojo
+        "{0}".format("hello")
+        ```
+
+        value field contains `0`
+
+    - `NoneType` for automatic indexing of *args
+
+        ```mojo
+        "{}".format("hello")
+        ```
+
+        value field contains `None`
+
+    `*_curly` are the indexes of the enclosing curlies
+
+    This struct only supports theses:
+
+    `{number|name|nothing}`
+
+    """
+
+    var first_curly: Int
+    var last_curly: Int
+    var value: Variant[String, Int, NoneType]
+
+    @staticmethod
+    fn create_entries(arg: String) -> List[Self]:
+        """Used internally by the `format()` method.
+
+        Args:
+            arg: The "format" part provided by the user.
+
+        Returns:
+            A `List` of structured format entries.
+        """
+        var Entries = List[Self]()
+        var start = Optional[Int](None)
+        for i in range(len(arg)):
+            if arg[i] == "{":
+                start = i
+                continue
+            if arg[i] == "}":
+                if start:
+                    var start_value = start.value()[]
+                    var tmp = Self(start_value, i, None)
```

Comment:
Here and elsewhere: Can you please come up with more meaningful variable names? For example `tmp` here is something like `current_entry`, right?

Reply:
Yep, that makes sense lol